source/common/http/http2/codec_impl.cc

```diff
@@ -627,8 +626,7 @@ int ConnectionImpl::onFrameSend(const nghttp2_frame* frame) {
     if (stream->headers_) {
       // Verify that the final HeaderMap's byte size is under the limit before sending frames.
       // This assert iterates over the HeaderMap.
-      ASSERT(stream->headers_->byteSize().has_value() &&
-             stream->headers_->byteSize().value() == stream->headers_->byteSizeInternal());
+      ASSERT(stream->headers_->byteSize() == stream->headers_->byteSizeInternal());
```

Comment:
Are these even needed anymore? This sees like an internal invariant in the header map.

Reply:
Yah, I commented about this. I removed them, wasn't sure if we wanted to keep them.

Reply:
I would nuke the internal asserts that are outside header map, and ensure there are sufficient internal invariant asserts for the same property.

Reply:
Done.